File: pset6/readability/readability.py
from cs50 import get_string

text = get_string("Text: ")

letters = symbols = words = 0

for char in text:
    if char.isalpha():
        letters += 1
    if char.isspace():
        words += 1
    if char in ['?', '.', '!']:
        symbols += 1

words += 1
L = (letters * 100.0) / words
S= (symbols * 100.0) / words
result = int((0.0588 * L - 0.296 * S - 15.8) + 0.5)

if result < 1:
    print("Before Grade 1")
elif result >= 16:
    print("Grade 16+")
else:
    print(f"Grade {result}")

File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>
#include <string.h>
#include <ctype.h>

int get_index(string s);

int main()
{
    string text = get_string("Text: ");
    int index = get_index(text);

    if(index < 1)
    {
        printf("Before Grade 1\n");
    }else if(index >= 16)
    {
        printf("Grade 16+\n");
    }else
    {
        printf("Grade %d\n", index);
    }

    return 0;
}

int get_index(string s)// Counting the letters, sentences and words
{
    int letters = 0, sentences = 0, words = 0;

    for(int i = 0; i <strlen(s); i++)
    {
        char ch = s[i];

        if(isalpha(ch))
        {
            letters++;
        }

        //hello world
        if(isspace(ch))
        {
            words++;
        }

        if(ch == '.' || ch == '!' || ch == '?')
        {
            sentences++;
        }
    }
    words++;

    float L = (letters * 100.0f) / words;
    float S = (sentences * 100.0f) / words;
    return round(0.0588 * L - 0.296 * S - 15.8);
}